src/compiler/base/nodes/tags/step.ts

```diff
@@ -80,7 +71,12 @@ export async function compile(
     }
 
     const textVarValue = (stepResponse?.content ?? []).filter(c => c.type === ContentType.text).map(c => c.text).join('')
-    scope.set(String(textVarName), textVarValue)
+
+    if ("schema" in config) {
+      scope.set(String(textVarName), JSON.parse(textVarValue.trim()))
```

Comment:
Try / catch this. Can fail

Reply:
We want to fail

Reply:
But I could throw a promptl error instead of json error maybe

Reply:
;)